File: WoofAndroid/design-system/src/main/kotlin/com/perrystreet/woof/designsystem/atomic/templates/TemplateCenteredContent.kt
package com.perrystreet.woof.designsystem.atomic.templates

import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.WindowInsets
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.safeDrawing
import androidx.compose.material3.Scaffold
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import com.perrystreet.woof.designsystem.atomic._tokens.spacing.PaddingRoles
import com.perrystreet.woof.designsystem.theme.Theme

@Composable
fun TemplateCenteredContent(
    topBar: @Composable () -> Unit,
    content: @Composable () -> Unit,
) {
    Scaffold(
        containerColor = Theme.colors.background,
        contentWindowInsets = WindowInsets.safeDrawing,
        topBar = {
            Box(modifier = Modifier.padding(horizontal = PaddingRoles.Screen.Regular.dp)) {
                topBar()
            }
        },
    ) { paddingValues ->
        Box(
            modifier = Modifier
                .fillMaxSize()
                .padding(paddingValues)
                .padding(horizontal = PaddingRoles.Screen.Regular.dp),
            contentAlignment = Alignment.Center,
        ) {
            content()
        }
    }
}

File: WoofIOS/SwiftPackages/DesignSystem/Sources/DesignSystem/Atomic/Templates/TemplateCenteredContent.swift
import SwiftUI

public struct TemplateCenteredContent<TopBar: View, Content: View>: View {
    @Environment(\.theme) private var theme

    private let topBar: () -> TopBar
    private let content: () -> Content

    public init(@ViewBuilder topBar: @escaping () -> TopBar, @ViewBuilder content: @escaping () -> Content) {
        self.topBar = topBar
        self.content = content
    }

    public var body: some View {
        VStack(spacing: 0) {
            topBar()
                .padding(.horizontal, PaddingRoles.Screen.regular.rawValue)
            content()
                .padding(.horizontal, PaddingRoles.Screen.regular.rawValue)
                .frame(maxWidth: .infinity, maxHeight: .infinity)
        }
        .background(theme.colors.background.ignoresSafeArea())
    }
}
